tests/page-objects/admin/languages.wdio.page.js

```diff
@@ -0,0 +1,62 @@
+const utils = require('../../utils');
+
+const submitButton = () => $('div.form-actions button.btn.btn-primary');
+const addLanguageButton = () => $('span=Add new language');
+const addLanguageModal = () => $('body.ng-scope.modal-open');
+const languageCodeInput = () => $('[ng-model="language.code"]');
+const languageNameInput  = () => $('[ng-model="language.name"]');
+const languageSubmitButton  = () => $('a.btn.submit.ng-scope.ng-binding.btn-primary');
+const applicationLink = () => $('span=Application');
+const defaultLanguageDropdown = () => $('#locale');
+const outgoingLanguageDropdown = () => $('#locale-outgoing');
+
+const goToLanguagesTab = async () => {
+  await browser.url(utils.getAdminBaseUrl() + 'display/languages');
+  await browser.refresh();
+  await (await addLanguageButton()).waitForDisplayed();
+};
+
+const openAddLanguageModal = async () => {
+  await (await addLanguageButton()).click();
+  await (await addLanguageModal()).waitForDisplayed();
+};
+
+const addNewLanguage = async (code, name) =>{
+  await (await addLanguageButton()).click();
+  await (await addLanguageModal()).waitForDisplayed();
+  await (await languageCodeInput()).waitForDisplayed();
+  await (await languageCodeInput()).setValue(code);
+  await (await languageNameInput()).setValue(name);
+  await (await languageSubmitButton()).click();
+  await (await languageSubmitButton()).waitForDisplayed({reverse:true});
+};
+
+const languageDisplayed = async (code) =>{
+  const languageDiv = () => $(`#locale-${code}`);
+  const languageName = await (await languageDiv()).getText();
+  return languageName;
+};
+
+const selectLanguage = async (element, code) => {
+  await (await element()).selectByAttribute('value', `string:${code}`);
+  await (await submitButton()).click();
+  await browser.waitUntil(async () => await (await $('.success.ng-binding.ng-scope')).getText() === 'Saved');
```

Comment:
`.ng-binding.ng-scope`  are both added by Angular on almost every element that is scoped and is bound, so it doesn't refine the selector. I think it's safe to remove them. Is there no other markup that can refine this selector?